### PR TITLE
docker:go builder: move PreModDownload hook

### PR DESCRIPTION
**Pull request by nonsense**

### CODE REVIEW

File: pkg/build/docker_go.go [hunk 1]
```diff
@@ -371,17 +371,17 @@ ARG BUILD_TAGS
 ENV TESTPLAN_EXEC_PKG ${TESTPLAN_EXEC_PKG}
 
 # PLAN_DIR is the location containing the plan source inside the container.
-ENV PLAN_DIR /plan/
+ENV PLAN_DIR /plan
+
+{{.DockerfileExtensions.PreModDownload}}
```
Comment (nonsense) [inline pkg/build/docker_go.go:376]: @raulk It'd be nice if we can install packages as part of the testplan ala https://github.com/filecoin-project/lotus-testground/blob/master/manifest.toml#L12 that don't get invalidated every time we touch the `go.mod` from the testplan. WDYT?

File: pkg/build/docker_go.go [hunk 2]
```diff
@@ -371,17 +371,17 @@ ARG BUILD_TAGS
 ENV TESTPLAN_EXEC_PKG ${TESTPLAN_EXEC_PKG}
 
 # PLAN_DIR is the location containing the plan source inside the container.
-ENV PLAN_DIR /plan/
+ENV PLAN_DIR /plan
```
Comment (nonsense) [inline pkg/build/docker_go.go:374]: No trailing slash for the default `PLAN_DIR`, which makes it more intuitive to use in the following hooks.

File: pkg/build/docker_go.go (hunk 1 above)
Comment (raulk) [reply pkg/build/docker_go.go:376]: I don't expect us to modify the `go.mod` of a test plan very frequently, but I also don't see a reason why this shouldn't be merged.

**Comment by raulk** [comment]
Needs rebasing on top of #1078 once that is merged.

File: pkg/build/docker_go.go (hunk 1 above)
Comment (nonsense) [reply pkg/build/docker_go.go:376]: Adding/removing debug packages and iterating on a testplan does hit `go.mod` often, resulting in invalidated cache.

**Pull request merged**